src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -192,18 +250,14 @@ RadixContainer<T> materialize_input(const std::shared_ptr<const Table>& in_table
 Build all the hash tables for the partitions of the build column. One job per partition
 */
 
-// For semi and anti joins, we only care whether a value exists or not, so there is no point in tracking the position
-// in the input table of more than one occurrence of a value. However, if we have secondary predicates, we do need to
-// track all occurrences of a value as that first position might be disqualified later.
-enum class JoinHashBuildMode { AllPositions, SinglePosition };
-
-template <typename BuildColumnType, typename HashedType, JoinHashBuildMode mode>
-std::vector<std::optional<HashTable<HashedType>>> build(const RadixContainer<BuildColumnType>& radix_container) {
+template <typename BuildColumnType, typename HashedType>
```

Comment:
You moved the mode template parameter into the map, which isn't templated afaik. Can this lead to performance regressions?

Reply:
Might be a small regression. Otherwise, however, we would have to pass the `JoinHashBuildMode` to the probe method so that it has the complete type of the hash table.

Overall, it's a performance improvement, so I wouldn't worry too much about it.

Reply:
Well, would it hurt to add them?

Reply:
```
+----------------+----------------+------+----------------+------+------------+-----------------------------------------------+
| Benchmark      | prev. iter/s   | runs | new iter/s     | runs | change [%] |               p-value (significant if <0.001) |
+----------------+----------------+------+----------------+------+------------+-----------------------------------------------+
| TPC-H 1        | 0.42155122757  | 5    | 0.424118191004 | 5    | +1%        | (run time too short) (not enough runs) 0.8167 |
| TPC-H 2        | 6.76054811478  | 68   | 6.78727293015  | 68   | +0%        |                   (run time too short) 0.3267 |
| TPC-H 3        | 6.29697704315  | 63   | 6.23570537567  | 63   | -1%        |                   (run time too short) 0.3636 |
| TPC-H 4        | 2.1995742321   | 22   | 2.16156339645  | 22   | -2%        |                   (run time too short) 0.0257 |
| TPC-H 5        | 3.89468789101  | 39   | 3.87368965149  | 39   | -1%        |                   (run time too short) 0.3702 |
| TPC-H 6        | 21.4303569794  | 215  | 21.4652404785  | 215  | +0%        |                   (run time too short) 0.7495 |
| TPC-H 7        | 1.44852221012  | 15   | 1.4402230978   | 15   | -1%        |                   (run time too short) 0.5963 |
| TPC-H 8        | 5.24962186813  | 53   | 5.34621810913  | 54   | +2%        |                   (run time too short) 0.0009 |
| TPC-H 9        | 1.40513062477  | 15   | 1.40292215347  | 15   | -0%        |                   (run time too short) 0.8628 |
| TPC-H 10       | 2.79722976685  | 28   | 2.82507348061  | 29   | +1%        |                   (run time too short) 0.0504 |
| TPC-H 11       | 22.526102066   | 226  | 22.587097168   | 226  | +0%        |                   (run time too short) 0.2477 |
| TPC-H 12       | 7.12229681015  | 72   | 7.03500747681  | 71   | -1%        |                   (run time too short) 0.0319 |
| TPC-H 13       | 2.22776532173  | 23   | 2.23344063759  | 23   | +0%        |                   (run time too short) 0.7584 |
| TPC-H 14       | 21.0300235748  | 211  | 21.192483902   | 212  | +1%        |                   (run time too short) 0.0062 |
| TPC-H 15       | 13.6221647263  | 137  | 14.0764980316  | 141  | +3%        |                   (run time too short) 0.0000 |
| TPC-H 16       | 6.74990177155  | 68   | 6.66032123566  | 67   | -1%        |                   (run time too short) 0.0003 |
| TPC-H 17       | 1.10464942455  | 12   | 1.13597083092  | 12   | +3%        |                   (run time too short) 0.0068 |
| TPC-H 18       | 1.4168368578   | 15   | 1.33194899559  | 14   | -6%        |                   (run time too short) 0.0097 |
| TPC-H 19       | 5.45627164841  | 55   | 5.4262046814   | 55   | -1%        |                   (run time too short) 0.0735 |
| TPC-H 20       | 1.89534187317  | 20   | 1.82846522331  | 19   | -4%        |                   (run time too short) 0.0131 |
| TPC-H 21       | 0.351791858673 | 4    | 0.353695809841 | 4    | +1%        | (run time too short) (not enough runs) 0.8130 |
| TPC-H 22       | 10.7620391846  | 108  | 10.8397464752  | 109  | +1%        |                   (run time too short) 0.0001 |
| geometric mean |                |      |                |      | -0%        |                                               |
+----------------+----------------+------+----------------+------+------------+-----------------------------------------------+
```

At least it doesn't bring anything to the table. Less templates are generally a good thing.